Sources/SemanticIndex/SemanticIndexManager.swift

```diff
@@ -209,11 +209,16 @@ public final actor SemanticIndexManager {
     for uri in changedFiles {
       indexStatus[uri] = nil
     }
-    // Clear the preparation status so that we re-prepare them. If the target hasn't been affected by the file changes,
-    // we rely on a null build during preparation to fast re-preparation.
-    // Ideally, we would have more fine-grained dependency management here and only mark those targets out-of-date that
-    // might be affected by the changed files.
-    preparationStatus.removeAll()
+    let targets = await changedFiles.asyncMap { await buildSystemManager.configuredTargets(for: $0) }.flatMap { $0 }
+    if let dependentTargets = await buildSystemManager.targets(dependingOn: targets) {
+      for dependentTarget in dependentTargets {
+        preparationStatus[dependentTarget] = nil
+      }
+    } else {
```

Comment:
The dependent targets won't include the target itself, right? Should we be setting that to `nil` too?

Reply:
A change to the target itself shouldn't re-trigger preparation for that target - we have those files in memory on the sourcekit side. Hence my other comment about:

> We could not remove targets in this case, not that it's really expected to happen anyway.

Even if we don't know what the dependencies are, we know that there's no need to prepare the target the file is in.